tests/client/amo/components/TestAddonReview.js

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import { findDOMNode } from 'react-dom';
+import {
+  findRenderedComponentWithType,
+  renderIntoDocument,
+  Simulate,
+} from 'react-addons-test-utils';
+
+import { setReview } from 'amo/actions/reviews';
+import * as amoApi from 'amo/api';
+import * as coreApi from 'core/api';
+import {
+  mapDispatchToProps, mapStateToProps, AddonReviewBase,
+  loadAddonReview as defaultAddonReviewLoader,
+} from 'amo/components/AddonReview';
+import I18nProvider from 'core/i18n/Provider';
+import { fakeAddon, signedInApiState } from 'tests/client/amo/helpers';
+import { getFakeI18nInst, RouterStub } from 'tests/client/helpers';
+
+const defaultReview = {
+  id: 3321, addonSlug: fakeAddon.slug,
+};
+
+function render({ fakeRouter = {}, ...customProps } = {}) {
+  const props = {
+    i18n: getFakeI18nInst(),
+    apiState: signedInApiState,
+    review: defaultReview,
+    updateReviewText: () => {},
+    ...customProps,
+  };
+  const root = findRenderedComponentWithType(renderIntoDocument(
+    <I18nProvider i18n={props.i18n}>
+      <RouterStub router={fakeRouter}>
```

Comment:
If you used `withRouter()` on the component as I mentioned above I think you might be able to pull out the `I18nProvider` and `RouterStub` components.

```jsx
<AddonReviewBase i18n={i18n} router={fakeRouter} {...props} />